src/pipeline.rs

```diff
@@ -1,18 +1,54 @@
 use std::fs::File;
 use std::io::{BufRead, BufReader};
 
-use crate::configuration::Configuration;
+use crate::configuration::{Column, Configuration, FileType};
 use crate::embedding::{calculate_embeddings, calculate_embeddings_mmap};
 use crate::entity::{EntityProcessor, SMALL_VECTOR_SIZE};
 use crate::persistence::embedding::TextFileVectorPersistor;
 use crate::persistence::entity::InMemoryEntityMappingPersistor;
 use crate::persistence::sparse_matrix::InMemorySparseMatrixPersistor;
 use crate::sparse_matrix::{create_sparse_matrices, SparseMatrix};
 use bus::Bus;
-use smallvec::SmallVec;
+use simdjson_rust::dom;
+use smallvec::{smallvec, SmallVec};
 use std::sync::Arc;
 use std::thread;
 
+/// Parse a line of JSON and read its columns into a vector for processing.
+fn read_json_columns(
```

Comment:
Could you move that function after `pub fn build_graphs`?

Reply:
Done.